### PR TITLE
Set, change and clear a node's scheduled time from the date picker

### PR DESCRIPTION
**Pull request by srid**
The Scheduled-for panel under a row could only pick a **day**. For a node scheduled for a time of day, it showed only the day and warned that picking one would drop the time. So a person couldn't set, change or clear a time, even though `outlines_date` (the agent's tool) can.

## What changes

- **A time box sits beside the day** (`<input type="time">`, for the same reason the day box is the browser's own control: it can't create an instant). Leave it empty and the node is scheduled for the day.
- **A day plus a time is saved as one instant**, written the way a mark's timestamp is: `2026-09-01T09:30:00-04:00`. The offset is the browser's offset *at that moment*, not today's, so December gets `-05:00`.
- **Change:** a timed node opens with both boxes filled. Changing either box saves the new value. If neither box changed, nothing is saved, so a stored value's seconds and offset stay as they are.
- **Clear the time:** **No time** empties the time box, and the button then says **Clear time**. Pressing it keeps the day. It also gives a label when a browser has half-emptied the time box.
- **Clear the date:** emptying the day box is still **Clear date**, and it clears the time too (a time with no day isn't a valid `date`).
- **Another zone:** a stored time with a different offset, or no offset, looks the same in the boxes. So the panel says which offset a changed day or time will be saved with.

The rules live in `date/pick.ts` and are pure functions of strings. The only use of `Date` is `browserOffsetAt`, which is passed in as an argument, so the unit tests fix a zone. `@olai/format` now also exports `timeOf` and `offsetOf`.

## Cordis / ownership

No ownership changes. The time draft is a second signal in the row's existing form memory (`date/memory.tsx`), next to the day draft. It has the same owner and lifetime, so it survives plugin rebuilds and pane switches and goes away on cancel or collapse, just like the day. The save still goes through the same `date` edit and write gate.

## Tests

- Unit: `pick.test.ts` rewritten for starting values, the saved value (unchanged time kept as-is, the offset of that moment, no day means no date), button labels and the notice.
- E2E (`setting_a_date.feature`): set a day and time, including a later date in a different offset; change a stored time; clear the time and undo with ⌘Z; empty the day on a timed node; a time alone is nothing to save; a stored time from another zone. `row_forms_lifecycle.feature`: a time draft survives a plugin change.
- New `@zone:<IANA>` tag sets the scenario's browser `timezoneId`, so offset checks don't depend on the machine running the tests (documented in `packages/tests/README.md`).
- Remote typecheck, unit tests and full e2e all passed on this tree.

Docs: `docs/editing.md` § Putting a node on a day.

### CODE REVIEW

File: packages/plugins/outlines/src/browser/date/pick.ts [hunk 1]
```diff
@@ -1,55 +1,115 @@
...
+ * written is still the day and the time the boxes hold.
+ *
+ * A time a clock skips (half past two on the morning it goes forward) is read
+ * with the offset `Date` lands on for it, which is what the platform does with
+ * that moment everywhere else in the tab.
+ */
+export const browserOffsetAt: OffsetAt = (day, time) => {
+  const [year, month, date] = day.split("-").map(Number) as [number, number, number]
+  const [hours, minutes] = time.split(":").map(Number) as [number, number]
+  return offsetOf(new Date(year, month - 1, date, hours, minutes).getTimezoneOffset())
```
Comment (srid) [inline packages/plugins/outlines/src/browser/date/pick.ts]: [P2] Resolve the whole instant consistently across a DST gap

In America/New_York, choosing `2026-03-08` and `02:30` produces `2026-03-08T02:30:00-04:00`. That instant is **01:30 EST** in the browser zone. The Date constructor normalized the nonexistent input forward to **03:30 EDT**, but only its offset survives here, so combining that offset with the original time moves the saved instant an hour backward instead. This contradicts the stated platform-compatible handling and the promise to write in the browser zone at that moment. Handle nonexistent local times explicitly (or normalize the whole date/time consistently), and add a transition-day test using the actual browserOffsetAt rather than the month-only offset stub.

File: packages/plugins/outlines/src/browser/date/DatePicker.tsx [hunk 2]
```diff
@@ -116,6 +129,34 @@ export function DatePicker(props: {
...
+          value={props.time}
+          onInput={(event) => props.onTime(event.currentTarget.value)}
+        />
+      </label>
+      {/* A way to empty the time box that every browser draws the same. The
+          platform's own time control has none on most engines — its fields are
+          emptied one at a time with Backspace — and taking a time off is one
+          of the three things this panel is for. It edits the draft only; the
+          button, which then says `Clear time`, is still what writes. */}
+      <Show when={props.time !== ""}>
```
Comment (srid) [inline packages/plugins/outlines/src/browser/date/DatePicker.tsx]: [P2] Keep a clearing control available for incomplete time entry

Start with an empty time box, pick a different day, then focus the time box and press ArrowUp to fill only its hour segment. Chromium reports `value === ""` and `validity.badInput === true`. Since the time signal was already empty, onInput does not trigger a value reset; this Show also hides `No time`. The enabled `Set date` button then cannot submit because RowPanel uses a native validating form, even though valueOf treats this draft as a valid day-only choice. Keep an explicit clear action reachable for this state and make it clear the actual input even when the signal is already empty, or otherwise distinguish incomplete input from an intentionally absent time. Add a keyboard-entry e2e scenario; fill() and the current No time scenario bypass this state. Native incomplete-input validation is specified in the [HTML time-input rules](https://html.spec.whatwg.org/multipage/input.html#time-state-(type=time)).

File: packages/plugins/outlines/src/browser/date/pick.ts [hunk 3]
```diff
@@ -62,43 +122,55 @@ export const startsAt = (stored: string | undefined): string =>
...
  */
-export const noticeOf = (stored: string | undefined): string | undefined =>
-  stored === undefined || stored === dayOf(stored)
+export const noticeOf = (
+  stored: string | undefined,
+  offsetAt: OffsetAt,
+): string | undefined => {
+  const time = stored === undefined ? undefined : timeOf(stored)
+  if (stored === undefined || time === undefined) return undefined
+  const here = offsetAt(dayOf(stored), time)
```
Comment (srid) [inline packages/plugins/outlines/src/browser/date/pick.ts]: [P2] Calculate the advertised save offset from the draft

With browser zone America/New_York and stored date `2026-08-10T14:30:00-07:00`, the notice says a changed day/time will be written with `-04:00`. Change the day to `2026-12-01`: the notice still promises `-04:00`, but valueOf writes `2026-12-01T14:30:00-05:00`. noticeOf only receives the stored value, so it cannot describe the offset of the pending edit. Pass the chosen day/time into the notice calculation and test changing across the seasonal boundary while a foreign-offset notice is visible.

**Review by srid — COMMENTED** [review]
Found three issues in the time-picker edge cases, detailed inline. The new draft follows the existing row-form ownership and write path. Validation: inspected all changed files and surrounding form/lifecycle code; executed the date conversion functions extracted from this revision under America/New_York; reproduced incomplete native time-input validation in Chromium with the same signal equality behavior. I did not run the full application e2e suite. The added scenarios cover ordinary seasonal offsets and explicit clearing, but miss DST gaps, incomplete keyboard entry, and the offset notice after changing seasons.

File: packages/plugins/outlines/src/browser/date/pick.ts (hunk 1 above)
Comment (srid) [reply packages/plugins/outlines/src/browser/date/pick.ts]: Fixed in fa8b762a7. The resolver now returns the whole instant instead of just an offset (`InstantAt` / `browserInstantAt`). It builds one `Date` from the typed day and time and writes it with `stampOf`, so the time and offset both come from the same moment. `2026-03-08` + `02:30` in New York is now saved as `2026-03-08T03:30:00-04:00`. Before pressing, the panel shows: "There is no 02:30 on 2026-03-08 in this browser's time zone, so pressing writes 2026-03-08T03:30:00-04:00." The year is set with `setFullYear`, so years below 100 aren't read as 19xx.

Tests: unit tests call the real `browserInstantAt` under `inZone("America/New_York")` for the skipped hour (checked to land after 01:59 EST), a repeated hour (time and offset stay consistent), a December offset and year 0050. New e2e scenario (`@zone:America/New_York`): "A time the clock skips is written as the moment it becomes, and says so first".

File: packages/plugins/outlines/src/browser/date/DatePicker.tsx (hunk 2 above)
Comment (srid) [reply packages/plugins/outlines/src/browser/date/DatePicker.tsx]: Fixed in fa8b762a7. The picker now reads `validity.badInput` from both boxes on `input` and on `keyup`. `keyup` catches the ArrowUp case, where one segment fills and the value stays empty. While either box is half-typed:
- the draft is left as it was. Writing the empty value into it would make Solid push `""` back into the box and wipe the segments still on screen. That's the Backspace-on-`14:30` case, which the first e2e run caught;
- the button is disabled, keeps the label "Set date", and the panel says "The time is not finished. Finish it, or press No time." Nothing reaches the browser's own form validation;
- **No time** stays visible, and it clears the input element itself as well as the draft.

The day box gets the same guard ("The day is not finished. Finish it, or empty it.").

E2E, using real key presses rather than `fill`: "A half-typed time from the keyboard writes nothing until it is finished or taken off" (ArrowUp on an empty time box) and "Emptying one segment of a stored time is not taking the time off" (Backspace on a stored time). "the date picker holds X at Y" now also checks `badInput === false`.

File: packages/plugins/outlines/src/browser/date/pick.ts (hunk 3 above)
Comment (srid) [reply packages/plugins/outlines/src/browser/date/pick.ts]: Fixed in fa8b762a7. `noticeOf` now takes the draft (`day`, `time`, `incomplete`) and works out the value pressing would save:
- stored time from another zone, unchanged: "…A changed day or time is written in this browser's time zone." It no longer names an offset, since that depends on a day and time not chosen yet;
- stored time from another zone, changed: it quotes the exact value, e.g. "Scheduled for 2026-08-10T14:30:00-07:00. Pressing writes 2026-12-01T14:30:00-05:00, in this browser's time zone.";
- clearing the time saves no offset, so there's no notice.

The e2e scenario "A time written in another zone says so, and quotes what a change writes" now moves the day to December while the notice is visible, checks the `-05:00` sentence, and checks the saved value. Unit tests cover the same cases.

**Comment by srid** [comment]
Addressed all three review comments in fa8b762a7 (details in each thread):

1. **DST gap:** the time and offset now come from one reading of the moment, so a skipped time is saved as the moment it becomes, and the panel quotes that value before saving.
2. **Half-typed time or day:** the button is disabled and the panel explains why. **No time** stays available and clears the actual input. The draft isn't overwritten, so a single Backspace no longer wipes the box.
3. **Offset in the notice:** it's worked out from the draft and quotes exactly what pressing would save.

`docs/editing.md` is updated to match.

Validation on this tree: remote typecheck, unit tests (including real `America/New_York` DST cases through `inZone`) and the full e2e suite all passed. New or changed e2e scenarios: DST gap, keyboard ArrowUp on an empty time box, Backspace on a stored time, and the foreign-offset notice across a season change. One e2e run lost all shards to runners dropping out and was re-run clean. An earlier run had failed only on the Backspace scenario; that exposed the draft-reset bug fixed here.

**Review by srid — COMMENTED** [review]
Re-reviewed at fa8b762a7ce61ae3e57a0a8133ce9e75287c8ee0. All three findings from my previous review are resolved:

- **DST-gap instant — resolved.** `browserInstantAt` now resolves and stamps the whole instant. The New York `2026-03-08 02:30` case saves `2026-03-08T03:30:00-04:00`, with a notice quoting that value before submission. The new unit tests exercise the real resolver, and a new e2e scenario checks both the notice and saved value.
- **Incomplete time entry — resolved.** The picker tracks native `badInput`, disables submission with an explanation, and keeps **No time** available. Clearing resets the DOM input as well as the draft, including when the draft was already empty. Partial deletion of a stored time is also guarded without erasing the remaining segments. Real-keyboard e2e scenarios cover both entry paths.
- **Stale offset notice — resolved.** `noticeOf` now receives the draft and quotes the exact pending value. Moving the foreign-offset August value to December correctly advertises and saves `2026-12-01T14:30:00-05:00`; the revised e2e scenario covers this transition.

**No new actionable findings.** I reviewed the follow-up diff, surrounding submission/lifecycle code, documentation, and regression coverage. The complete day/time draft remains with the existing row-form owner; transient native-input validity stays with the mounted controls.

Validation: `just test-fast-remote` passed on this head (Odu run `0mtzykhbc-c19xzrci`, all six test shards passed). A targeted Chromium harness using this revision's extracted picker rules/handlers and the actual Solid signal runtime also passed incomplete-time clearing, completing a partially cleared stored time, partial-day guarding/clearing, DST-gap conversion, and the December notice/save case. That harness is not a full application e2e run; I inspected the added e2e scenarios but did not rerun the full e2e suite.
